src/MetadataManager/PathToHash.ts

```diff
@@ -0,0 +1,293 @@
+ */
+
+class SyncMetadata{
+  static async validateMetadata(flattenMap: any) {
+
+    /**  1. Create metadata if pathToHash exists but does not have actual metadata files,
+     *      If pathToHash exists and there is a actual metadata file, but there is no path inside, create
+     *      a path and data insde
+     */
+    await this.createMetadata(flattenMap);
+
+    // 2. Replace is_deleted with true for all metadata not in pathToHash
+    await this.deleteMetadata(flattenMap);
+  }
+
+  static async createMetadata(flattenMap: any) {
+    for (let path in flattenMap) {
+      const hash = flattenMap[path];
+
+
+      // It will probably be replaced by the code of the Metadata module from here
+      // maybe Metadata.create(hash)
+      let metadata: any = await Metadata.get(hash);
+
+      if (Object.keys(metadata).length === 0) {
+        await Metadata.set(hash, {});
+
+      } else if (metadata[path]) {
+        return;
+      }
+
+      const filename: any = path.split('/').pop();
+      if (vscode.workspace.workspaceFolders === undefined) {
+        return;
+      }
+      const stats: any = await Metadata.getStats(
+          vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri, path));
+      metadata[path] = {};
+      metadata[path]['name'] = filename;
+      metadata[path]['file-extension'] = filename.split('.')[1];
+      metadata[path]['create-time'] = stats.birthtime;
+      metadata[path]['modified-time'] = stats.mtime;
+      metadata[path]['is-deleted'] = false;
+      await Metadata.set(hash, metadata);
+
+      // to here
+    }
+  }
+
+  static async deleteMetadata(flattenMap: any) {
+    if (vscode.workspace.workspaceFolders === undefined) {
+      return;
+    }
+    const baseUri =
+        vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri, '.meta/hash_objects');
+    const files = await vscode.workspace.fs.readDirectory(baseUri);
+    for (const file of files) {
+      const hashFolderUri = vscode.Uri.joinPath(baseUri, file[0]);
+      const hashList = await vscode.workspace.fs.readDirectory(hashFolderUri);
+      for (const hashFile of hashList) {
+        const hashUri = vscode.Uri.joinPath(hashFolderUri, hashFile[0]);
+        let metadata =
+            JSON.parse(Buffer.from(await vscode.workspace.fs.readFile(hashUri)).toString());
+        const hash = file[0] + hashFile[0].split('.')[0];
+        for (const key in metadata) {
+          if (!metadata[key]['is-deleted'] && flattenMap[key] !== hash) {
+            metadata[key]['is-deleted'] = true;
+          }
+        }
+        Metadata.set(hash, metadata);
+      }
+    }
+  }
+}
+
+export class PathToHash {
+  private static _instance: PathToHash;
+  private _map: any;
+
+  private constructor() {}
+
+  public static async getInstance() {
+    if (!this._instance) {
+      this._instance = new PathToHash();
+      this._instance._map = await this._instance.init();
+      await SyncMetadata.validateMetadata(this._instance.getFlatMap());
```

Comment:
```suggestion
      await SyncMetadata.validateMetadata(await this._instance.getFlatMap());
```

Reply:
Oh, I'll reflect this on the next commit, thank you